### DISCUSSION TITLE
Broadcaster Memory Backend not working with Daphne

### DISCUSSION THREAD

**Discussion by fabiangfd** [encode/broadcaster#84] [Q&A]
I am having problems getting broadcaster to work with Daphne. Specifically, I would like to get this simple example to work:
[Ariadne GraphQL Subscription Example](https://ariadnegraphql.org/docs/subscriptions#simple-chat-example) (code below)
The app works as intended when run with Uvicorn. 

By adding print statements, I was able to deduce, that the `async for event in subscriber:`-loop is never entered though `broadcast.publish(...)` is executed. This is only the case when run with Daphne. When run with Uvicorn, the loop is entered as expected and the result for the subscription is returned.

Unfortunately, I am dependent on Daphne, so I cannot switch.

Any help would be greatly appreciated.  

Here is the code:

```
import json

from ariadne import MutationType, SubscriptionType, make_executable_schema
from ariadne.asgi import GraphQL
from ariadne.asgi.handlers import GraphQLWSHandler
from broadcaster import Broadcast
from starlette.applications import Starlette


broadcast = Broadcast("memory://")


type_defs = """
  type Query {
    _unused: Boolean
  }

  type Message {
    sender: String
    message: String
  }

  type Mutation {
    send(sender: String!, message: String!): Boolean
  }

  type Subscription {
    message: Message
  }
"""


mutation = MutationType()


@mutation.field("send")
async def resolve_send(*_, **message):
    await broadcast.publish(channel="chatroom", message=json.dumps(message))
    return True


subscription = SubscriptionType()


@subscription.source("message")
async def source_message(_, info):
    async with broadcast.subscribe(channel="chatroom") as subscriber:
        async for event in subscriber:
            # >>>> This is never executed when run with daphne:
            yield json.loads(event.message)
            # <<<<


schema = make_executable_schema(type_defs, mutation, subscription)
graphql = GraphQL(
    schema=schema,
    debug=True,
    websocket_handler=GraphQLWSHandler(),
)

app = Starlette(
    debug=True,
    on_startup=[broadcast.connect],
    on_shutdown=[broadcast.disconnect],
)

app.mount("/", graphql)
```

**Comment by Kludex** [comment]
...but why is the issue here and not on Daphne? :thinking:

**Reply by fabiangfd** [reply]
I know that it is probably a Daphne issue and I will post it there as well.  I was hoping someone here might already know a workaround.

**Reply by Kludex** [reply]
Ah, sure. :+1:

**Reply by fabiangfd** [reply]
For reference: here is the post on the daphne discussion board:
[https://github.com/django/daphne/discussions/459](https://github.com/django/daphne/discussions/459)
